src/coreclr/pal/src/exception/machexception.cpp

```diff
@@ -1261,14 +1277,29 @@ MachExceptionInfo::MachExceptionInfo(mach_port_t thread, MachMessage& message)
         Subcodes[i] = message.GetExceptionCode(i);
 
 #if defined(HOST_AMD64)
+    int cpuFeatures = minipal_getcpufeatures();
     mach_msg_type_number_t count = x86_THREAD_STATE_COUNT;
     machret = thread_get_state(thread, x86_THREAD_STATE, (thread_state_t)&ThreadState, &count);
     CHECK_MACH("thread_get_state", machret);
 
-    count = x86_FLOAT_STATE_COUNT;
-    machret = thread_get_state(thread, x86_FLOAT_STATE, (thread_state_t)&FloatState, &count);
-    CHECK_MACH("thread_get_state(float)", machret);
-
+    if (((cpuFeatures & XArchIntrinsicConstants_Avx512f) != 0))
+    {
+        count = x86_AVX512_STATE_COUNT;
+        machret = thread_get_state(thread, x86_AVX512_STATE, (thread_state_t)&AVX512State, &count);
+        CHECK_MACH("thread_get_state(avx512)", machret);
+    }
+    else if(((cpuFeatures & XArchIntrinsicConstants_Avx2) != 0))
+    {
+        count = x86_AVX_STATE_COUNT;
+        machret = thread_get_state(thread, x86_AVX_STATE, (thread_state_t)&AVXState, &count);
+        CHECK_MACH("thread_get_state(avx)", machret);
+    }
+    else
+    {
+        count = x86_FLOAT_STATE_COUNT;
+        machret = thread_get_state(thread, x86_FLOAT_STATE, (thread_state_t)&FloatState, &count);
+        CHECK_MACH("thread_get_state(float)", machret);
+    }
```

Comment:
@janvorli Are checks needed for this portion of the code? Or does it work to use lines 1287-1289?